frontend/src/components/ReadEditPage/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { Box, Button } from "@mui/material";
+import { CreateForm } from "components/CreateForm";
+import { EditForm } from "components/EditForm";
+import { ReadForm } from "components/ReadForm";
+import { Renderer } from "components/Renderer";
+import { FormikValues } from "formik";
+import { useFormSubmit } from "hooks/useFormSubmit";
+import React, { useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import { Navigate, useParams } from "react-router-dom";
+import { IEditFields, IWhat } from "types";
+import { apiAxios } from "utils";
+
+export const ReadEditPage = ({
+  what,
+  editFields,
+  readFields,
+  createFormInitialValues,
+  editCapability,
+  apiRoute,
+  redirectRoute,
+}: {
+  what: IWhat;
+  editFields: IEditFields[];
+  readFields: Function;
+  /**
+   * @todo Define a good type. "Any" type temporarily permitted.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  createFormInitialValues: any;
+  editCapability: string;
+  apiRoute: string;
+  redirectRoute: string;
+}) => {
+  const { id } = useParams();
+  const [editMode, setEditMode] = useState(false);
+  const [isNew, setNew] = useState(false);
+  const [createCompleted, setCreateCompleted] = useState(false);
+  const [userHasEditCapability, setEditCapability] = useState(false);
+  const { handlePost, handleUpdate, Notification } = useFormSubmit();
+
+  const getItem = async () => {
+    if ("new" === id) {
+      return null;
+    }
+    const item = await apiAxios().get(`${apiRoute}/${id}`);
+    return item.data;
+  };
+
+  /**
+   * @todo Define a good type. "Any" type temporarily permitted.
```

Comment:
again i'm pretty sure @todo don't get captured, like in PHP at current, maybe there is a jsdoc ext we need to add or add  a script to find all of these.

Reply:
How should we mark todos then? I'm just following the other places where we've added these `//todo find a better type` comments but I figure it's better use @todo than //todo since it seems like it would have a better chance of getting picked up even if it isn't right now.